tensorflow/contrib/seq2seq/python/kernel_tests/seq2seq_test.py

```diff
@@ -0,0 +1,121 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+"""Tests for contrib.seq2seq.python.ops.seq2seq."""
+# pylint: disable=unused-import,g-bad-import-order
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+# pylint: enable=unused-import
+
+import tensorflow as tf
+from tensorflow.contrib import layers
+
+class Seq2SeqTest(tf.test.TestCase):
+
+  # test a default call of rnn_decoder
+  def test_rnn_decoder(self):
+    pass
+
+  # test default call with time_major=True
+  def test_dynamic_rnn_decoder_time_major(self):
+    with self.test_session() as sess:
+      with tf.variable_scope("root", initializer=
+                             tf.constant_initializer(0.5)) as varscope:
+        # Define inputs/outputs to model
+        batch_size = 2
+        encoder_embedding_size = 3
+        decoder_embedding_size = 4
+        encoder_hidden_size = 5
+        decoder_hidden_size = encoder_hidden_size
+        input_sequence_length = 6
+        decoder_sequence_length = 7
+        num_decoder_symbols = 20
+        start_of_sequence_id = end_of_sequence_id = 1
+        decoder_embeddings = tf.get_variable('decoder_embeddings',
+            [num_decoder_symbols, decoder_embedding_size],
+            initializer=tf.random_normal_initializer(stddev=0.1))
+        inputs = tf.constant(0.5, shape=[input_sequence_length, batch_size,
+                                         encoder_embedding_size])
+        decoder_inputs = tf.constant(0.4, shape=[decoder_sequence_length,
+                                                 batch_size,
+                                                 decoder_embedding_size])
+        decoder_length = tf.constant(decoder_sequence_length, dtype=tf.int32,
+                                     shape=[batch_size,])
+        # setting up weights for computing the final output
+        output_fn = lambda x: layers.linear(x, num_decoder_symbols,
+                                            scope=varscope)
+
+        # Define model
+        encoder_outputs, encoder_state = tf.nn.dynamic_rnn(
+            cell=tf.nn.rnn_cell.GRUCell(encoder_hidden_size), inputs=inputs,
+            dtype=tf.float32, time_major=True, scope="rnn")
+
+        # Train decoder
+        decoder_cell = tf.nn.rnn_cell.GRUCell(decoder_hidden_size)
+        decoder_fn_train = tf.contrib.seq2seq.simple_decoder_fn_train(
+            encoder_state=encoder_state)
+        decoder_outputs_train, decoder_state_train = (
+          tf.contrib.seq2seq.dynamic_rnn_decoder(
+              cell=decoder_cell,
+              decoder_fn=decoder_fn_train,
+              inputs=decoder_inputs,
+              sequence_length=decoder_length,
+              time_major=True))
```

Comment:
this leads to an error:


    name, "".join(traceback.format_list(tb))))
ValueError: Variable root/rnn/gru_cell/gates/weights already exists, disallowed. Did you mean to set reuse=True in VarScope? Originally defined at:


due to a change i recently pushed.  this is because you're using the "rnn" scope twice.  what you should do is:

with tf.variable_scope("rnn") as scope:
  tf.nn.dynamic_rnn(..., scope=scope)

scope.set_reuse()
... = tf.contrib.seq2seq.dynamic_rnn_decoder(..., scope=scope)
...
... = tf.contrib.seq2seq.dynamic_rnn_decoder(..., scope=scope)